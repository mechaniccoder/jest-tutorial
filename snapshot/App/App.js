import React from "react";

const App = () => {
  const handle = () => {
    console.log("hi");
  };

  const fetch = () => {
    console.log("hi");
  };

  const hello = () => {
    console.log("try");
  };

  const say = () => {
    console.log("say");
  };

  return <div onClick={handle}>appewe TETW</div>;
};

export default App;
