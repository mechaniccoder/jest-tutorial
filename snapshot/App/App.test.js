import renderer from "react-test-renderer";
import App from "./App";
import React from "react";

it("renders app", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders app secondly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchInlineSnapshot(`
<div
  onClick={[Function]}
>
  appewe TETW
</div>
`);
});

it("will fail every time", () => {
  const user = {
    createdAt: new Date(),
    id: 10,
    name: "sh",
  };
  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: 10,
    name: expect.any(String),
  });
});
