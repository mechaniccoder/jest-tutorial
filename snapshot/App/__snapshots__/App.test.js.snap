// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<div
  onClick={[Function]}
>
  appewe TETW
</div>
`;

exports[`will fail every time 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": 10,
  "name": Any<String>,
}
`;
